### PR TITLE
Get rid of dependency on Boost

### PR DESCRIPTION
**Pull request by emaxx-google**
Remove all usages of the Boost C++ library from the project.

This allows to resolve the build issues caused by some problems
with the NaCl webport of Boost:

  /usr/bin/git-secrets: line 116: /usr/lib/git-core/git: Argument list too long

### CODE REVIEW

File: common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc [hunk 1]
```diff
@@ -59,8 +35,9 @@ bool VarAsInteger(
     if (!CastDoubleToInt64(var.AsDouble(), &integer, error_message))
       return false;
   } else {
-    *error_message = FormatBoostFormatTemplate(
-        kErrorWrongType, kIntegerJsTypeTitle, DebugDumpVar(var));
+    *error_message = FormatPrintfTemplate(
+        "Expected a value of type \"%s\", instead got: %s",
+        kIntegerJsTypeTitle, DebugDumpVar(var).c_str());
```
Comment (alex292) [inline common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc]: nit: formatting inconsistent with similar patterns below

File: common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc [hunk 2]
```diff
@@ -18,32 +18,8 @@
 
 #include <google_smart_card_common/numeric_conversions.h>
 
-// Error messages and related bits of strings
-const char kErrorWrongType[] =
```
Comment (alex292) [inline common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc]: this string is used multiple times below, so I would keep it defined as variable here

File: common/cpp/src/google_smart_card_common/pp_var_utils/extraction.h [hunk 3]
```diff
@@ -264,8 +257,10 @@ inline bool IsVarArraySizeValid(
     size_t expected_size,
     std::string* error_message) {
   if (var.GetLength() != expected_size) {
-    *error_message = FormatBoostFormatTemplate(
-        kErrorWrongArraySize, expected_size, var.GetLength());
+    *error_message = FormatPrintfTemplate(
+        "Expected an array of size %zu, instead got: %" PRIu32,
```
Comment (alex292) [inline common/cpp/src/google_smart_card_common/pp_var_utils/extraction.h]: %zu -> %PR... ?

Comment (emaxx-google) [reply common/cpp/src/google_smart_card_common/pp_var_utils/extraction.h]: Ack - the first parameter has the |size_t| type, and for it, I believe, the "zu" specifier should be used. The "PR..." thing is mainly for types with fixed size.

File: common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc (hunk 1 above)
Comment (emaxx-google) [reply common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc]: Done

File: common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc (hunk 2 above)
Comment (emaxx-google) [reply common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc]: Done

Comment (emaxx-google) [reply common/cpp/src/google_smart_card_common/pp_var_utils/extraction.cc]: Done (I was afraid that compiler won't be able to verify the printf format specifiers in that case, but looks like it's able to understand the constants defined in the same file as well)

File: common/cpp/src/google_smart_card_common/numeric_conversions.h [hunk 4]
```diff
@@ -96,12 +93,13 @@ inline bool CastInt64ToInteger(
...
-      type_name,
+  *error_message = FormatPrintfTemplate(
+      "The integer value is outside the range of type \"%s\": %" PRId64 " not "
+      "in [%" PRId64 "; %" PRIu64 "] range",
+      type_name.c_str(),
       value,
-      std::numeric_limits<T>::min(),
-      std::numeric_limits<T>::max());
+      static_cast<int64_t>(std::numeric_limits<T>::min()),
+      static_cast<uint64_t>(std::numeric_limits<T>::max()));
```
Comment (Fabian-Sommer) [inline common/cpp/src/google_smart_card_common/numeric_conversions.h]: nit: I think intmax_t and uintmax_t would be better here

Comment (emaxx-google) [reply common/cpp/src/google_smart_card_common/numeric_conversions.h]: Done.
Thanks for the tip! Didn't know about this thing.

**Comment by emaxx-google** [comment]
Thanks all!

**Pull request merged**